includes/data-stores/class-wc-webhook-data-store.php

```diff
@@ -0,0 +1,214 @@
+		global $wpdb;
+
+		$changes = $webhook->get_changes();
+		if ( isset( $changes['date_modified'] ) ) {
+			$date_modified     = $webhook->get_date_modified()->date( 'Y-m-d H:i:s' );
+			$date_modified_gmt = gmdate( 'Y-m-d H:i:s', $webhook->get_date_modified()->getTimestamp() );
+		} else {
+			$date_modified     = current_time( 'mysql' );
+			$date_modified_gmt = current_time( 'mysql', 1 );
+			$webhook->set_date_modified( $date_modified );
+		}
+
+		$data = array(
+			'status'            => $webhook->get_status( 'edit' ),
+			'name'              => $webhook->get_name( 'edit' ),
+			'user_id'           => $webhook->get_user_id( 'edit' ),
+			'delivery_url'      => $webhook->get_delivery_url( 'edit' ),
+			'secret'            => $webhook->get_secret( 'edit' ),
+			'topic'             => $webhook->get_topic( 'edit' ),
+			'date_modified'     => $date_modified,
+			'date_modified_gmt' => $date_modified_gmt,
+			'api_version'       => $this->get_api_version_number( $webhook->get_api_version( 'edit' ) ),
+			'failure_count'     => $webhook->get_failure_count( 'edit' ),
+			'pending_delivery'  => $webhook->get_pending_delivery( 'edit' ),
+		);
+
+		// @codingStandardsIgnoreStart
+		$wpdb->update(
+			$wpdb->prefix . 'woocommerce_webhooks',
+			$data,
+			array(
+				'webhook_id' => $webhook->get_id( 'edit' ),
+			)
+		);
+		// @codingStandardsIgnoreEnd
+
+		$webhook->apply_changes();
+
+		wp_cache_delete( $webhook->get_id(), 'webhooks' );
+		do_action( 'woocommerce_webhook_updated', $webhook->get_id() );
+	}
+
+	/**
+	 * Remove a webhook from the database.
+	 *
+	 * @since 3.2.0
+	 * @param WC_Webhook $webhook      Webhook instance.
+	 * @param bool       $force_delete Skip trash bin forcing to delete.
+	 */
+	public function delete( &$webhook, $force_delete = false ) {
+		global $wpdb;
+
+		// @codingStandardsIgnoreStart
+		$wpdb->delete(
+			$wpdb->prefix . 'woocommerce_webhooks',
+			array(
+				'webhook_id' => $webhook->get_id(),
+			),
+			array( '%d' )
+		);
+		// @codingStandardsIgnoreEnd
+
+		delete_transient( 'woocommerce_webhook_ids' );
+		do_action( 'woocommerce_webhook_deleted', $webhook->get_id(), $webhook );
+	}
+
+	/**
+	 * Get API version number.
+	 *
+	 * @since  3.2.0
+	 * @param  string $api_version REST API version.
+	 * @return int
+	 */
+	public function get_api_version_number( $api_version ) {
+		return 'legacy_v3' === $api_version ? -1 : intval( substr( $api_version, -1 ) );
+	}
+
+	/**
+	 * Get all webhooks IDs.
+	 *
+	 * @since  3.2.0
+	 * @return int[]
+	 */
+	public function get_webhooks_ids() {
+		global $wpdb;
+
+		$ids = get_transient( 'woocommerce_webhook_ids' );
+
+		if ( false === $ids ) {
+			// @codingStandardsIgnoreStart
+			$results = $wpdb->get_results( "SELECT webhook_id FROM {$wpdb->prefix}woocommerce_webhooks" );
+			// @codingStandardsIgnoreEnd
+			$ids = array_map( 'intval', wp_list_pluck( $results, 'webhook_id' ) );
+
+			set_transient( 'woocommerce_webhook_ids', $ids );
```

Comment:
I just copied it from `WooCommerce` class.
There is no need to expired if you don't change anything on your webhooks.